Refactored mentor listing page

kawinie opened the pull request:
# Purpose

This PR provides general code refactor and minor design changes in mentor listing page. 
Closes (#48)

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist

- [x] I have self-reviewed this PR
- [x] I have followed coding style guidelines of this project
- [x] I have not introduced new linting errors and warnings
- [x] I have commented part of the code that might be confusing or hard to understand
- [x] I have cleaned up WIP commits using Interactive Rebasing
- [x] I have rebased new changes into master and resolved any conflicts

# List of changes

- The mentor listing page is now broken down into components so that the code is easier to follow
- Removed mentor filter from components as there is no need for it at this time
- Added pages folder in components. This folder should be used exclusively for components specific to a page.
- Removed drawer functionality for filter sidebar on mobile devices

jettseale commented on pages/mentor-listing.tsx:
```diff
@@ -0,0 +1,64 @@
+export default function MentorListing() {
+    const { min } = useScreen();
+    return (
+        // Setting the height=100vh and overflow=hidden get rid of the scrollbar
```
You say here that it gets rid of the scrollbar, but when I look at the mentor listing page it still has the vertical scrollbar on the right-hand side. Do you mean the horizontal scrollbar?

ktsht replied:
I guess this is for the grey background box. We have a vertical scrollbar for the card list, which is fine, but if you scroll down to the bottom without this setting, another scrollbar shows up. I think that's what this deals with

ktsht replied:
This disables the scrollbar, but the lower parts of the bottommost cards are hidden😭

ktsht replied:
Thinking of the solution for it, but I don't get it now. Maybe we can create a minor PR at some point. It's not that big issue.

kawinie replied:
@ktsht What do you mean by "but the lower parts of the bottommost cards are hidden"?

jettseale reviewed (approved):
I'm sure this can be easily changed in another PR, but I noticed when scrolling the filter sidebar there is an overlap of text as shown in this screenshot: 
![Screenshot 2021-02-03 125145](https://user-images.githubusercontent.com/43707457/106807948-992f5480-661e-11eb-91c3-9f53737e101a.png)

Also, out of curiosity, why is the drawer functionality for the sidebar on mobile devices being removed? Surely we want mobile users to be able to use the filter as opposed to just getting rid of it?

kawinie commented:
@jettseale Wowwww thanks man. It works in Firefox but not others. Guess what? I fixed the Chrome version then Safari broke. I just spent 2 extra hours more to find fix for it. Finally solved it!!

I removed the mobile thing because I'm thinking of doing something like this. I want to integrate it with the navigation for the whole site.
<img width="348" alt="Screen Shot 2021-02-04 at 6 11 01 AM" src="https://user-images.githubusercontent.com/39807451/106821455-c37d2280-66af-11eb-8b0b-602422f3b7d8.png">

jettseale reviewed (approved):
Looking good! Great work.

ktsht commented on pages/mentor-listing.tsx:
```diff
@@ -0,0 +1,64 @@
+function Results() {
+    return (
+        <Box tw="h-screen w-full" overflow="scroll">
+            <Grid tw="justify-start items-center p-8 gap-x-4 gap-y-1">
```
Grid looks nicer and simpler than using VStack and HStack here. nice.

ktsht commented on pages/mentor-listing.tsx:
```diff
@@ -0,0 +1,64 @@
+        <Box tw="h-screen overflow-hidden" background="trueGray.100">
+            <NavBar name="John" />
+            <HStack alignItems="start">
+                {min`sm` && <FilterSidebar />}
```
Oh, we lost responsiveness here. Now mobile users cannot filter users. Jett made a button on the side of screen so that mobile users can use the filter feature.

ktsht replied:
I've just seen your conversation with Jett now. Nvm. We will add the bottom bar for mobile version, right?

kawinie replied:
Yeah. The previous solution was not fit for mobile so I'll putting that into our backlog.

ktsht reviewed (approved):
Good job~ I thought you were going to update the Nav search bar, since we are using material-ui here, but maybe we can do that in another PR

kawinie reviewed (commented):
> Good job~ I thought you were going to update the Nav search bar, since we are using material-ui here, but maybe we can do that in another PR

@ChibaTosh is working on the Navbar. He will need to rebase after this PR though.
